racket/collects/racket/private/dict.rkt

```diff
@@ -244,23 +196,39 @@
       (raise-argument-error 'dict-map "dict?" d))
     (proc (car x) (cdr x))))
 
+(define (assoc-fold-unique f init d)
+  (unless (assoc? d)
+    (raise-argument-error 'assoc-fold-unique "dict?" d))
```

Comment:
The `assoc?` check is repeated both here and in `assoc-count`. I think it should either always be done in the public function, or always done here, but using a `who` argument to report errors in terms of the public name.